
Following discussions with Yuqi:

- Obtain a better region than the Pu-27 with the following conditions:
  - Promoter (-1kb)
  - Strong BG4 signal (BG4 ChIP-seq) (+PQS G3L12)
  - Gene with elevated expression (RNA-seq) Herdy2018 paper
  - Open chromatin: DNase

- Create profile of ENCODE datasets to check around the selected promoter
- Create profile of JASPAR TF matches in the Pu-27 region of the MYC promoter

This is all in HEK293



## Datasets

### BG4 ChIP-seq

Given by Yuqi:

```bash
cd /scratchb/sblab/martin03/repository/20200217_yuqi/data/20200217/bg4
ls -lh
#-rwxrwxrwx 1 martin03 sblab 596K Jan 10  2019 HEK293_NTC_no_DOX_rep1.hg19.q005.all_peaks.multi2.bed

wc -l HEK293_NTC_no_DOX_rep1.hg19.q005.all_peaks.multi2.bed # 25558

sed 's/$/&\tBG4/' HEK293_NTC_no_DOX_rep1.hg19.q005.all_peaks.multi2.bed > HEK293_NTC_no_DOX_rep1.hg19.q005.all_peaks.multi2.mod.bed
```

However the above is an old dataset where peaks especially MYC is not strong enough.

Yuqi, Jochen, Angela have generated a new BG4 map in HEK293 and HeLa:

```bash
cd /scratchb/sblab/martin03/repository/20200217_yuqi/data/20200217/bg4

##########
# HEK293 #
##########
mkdir SLX-19000_Hek && cd SLX-19000_Hek

rsync -arvuP martin03@10.20.192.94:/mnt/nfs/nas/group_folders/simeon01/Data/20200205_yuqi_sequencing/SLX-19000_Hek/bw .
rsync -arvuP martin03@10.20.192.94:/mnt/nfs/nas/group_folders/simeon01/Data/20200205_yuqi_sequencing/SLX-19000_Hek/macs2_individual_rep .
cd macs2_individual_rep


# biorep1
wc -l Hek_biorep1*
  #  6890 Hek_biorep1_ChIP1_SLX-19000.i701_i502.r_1.hg38.q005.all_peaks.narrowPeak
  #  8584 Hek_biorep1_ChIP2_SLX-19000.i702_i502.r_1.hg38.q005.all_peaks.narrowPeak
  # 10165 Hek_biorep1_ChIP3_SLX-19000.i703_i502.r_1.hg38.q005.all_peaks.narrowPeak

bedtools multiinter -i Hek_biorep1* | \
awk -v OFS="\t" '$4 > 1 {print $0}' | \
bedtools sort -i - | \
bedtools merge -i - > Hek_biorep1.bed

wc -l Hek_biorep1.bed # 7931


# biorep2
wc -l Hek_biorep2*
  # 15969 Hek_biorep2_ChIP1_SLX-19000.i701_i503.r_1.hg38.q005.all_peaks.narrowPeak
  # 12110 Hek_biorep2_ChIP2_SLX-19000.i702_i503.r_1.hg38.q005.all_peaks.narrowPeak
  #  9884 Hek_biorep2_ChIP3_SLX-19000.i703_i503.r_1.hg38.q005.all_peaks.narrowPeak

bedtools multiinter -i Hek_biorep2* | \
awk -v OFS="\t" '$4 > 1 {print $0}' | \
bedtools sort -i - | \
bedtools merge -i - > Hek_biorep2.bed

wc -l Hek_biorep2.bed # 11665


# biorep3
wc -l Hek_biorep3*
  # 14524 Hek_biorep3_ChIP1_SLX-19000.i701_i504.r_1.hg38.q005.all_peaks.narrowPeak
  # 15506 Hek_biorep3_ChIP2_SLX-19000.i702_i504.r_1.hg38.q005.all_peaks.narrowPeak
  # 12541 Hek_biorep3_ChIP3_SLX-19000.i703_i504.r_1.hg38.q005.all_peaks.narrowPeak

bedtools multiinter -i Hek_biorep3* | \
awk -v OFS="\t" '$4 > 1 {print $0}' | \
bedtools sort -i - | \
bedtools merge -i - > Hek_biorep3.bed

wc -l Hek_biorep3.bed # 13049


# all biorep
wc -l Hek_biorep1.bed Hek_biorep2.bed Hek_biorep3.bed
 #  7931 Hek_biorep1.bed
 # 11665 Hek_biorep2.bed
 # 13049 Hek_biorep3.bed

bedtools multiinter -i Hek_biorep1.bed Hek_biorep2.bed Hek_biorep3.bed | \
awk -v OFS="\t" '$4 > 1 {print $0}' | \
bedtools sort -i - | \
bedtools merge -i - > Hek_narrow.bed

wc -l Hek_narrow.bed # 10745

tableCat.py -i Hek_biorep1.bed Hek_biorep2.bed Hek_biorep3.bed | \
awk -v OFS="\t" '{print $1, $2, $3, $NF}' | \
sort -k1,1 -k2,2n | \
bedtools merge -c 4,4 -o distinct,count_distinct -i - | \
awk -v OFS="\t" '$5 > 1 {print $1, $2, $3}' > Hek_broad.bed

wc -l Hek_broad.bed # 9337


########
# HeLa #
########
cd /scratchb/sblab/martin03/repository/20200217_yuqi/data/20200217/bg4

mkdir SLX-18996_HeLa && cd SLX-18996_HeLa

rsync -arvuP martin03@10.20.192.94:/mnt/nfs/nas/group_folders/simeon01/Data/20200205_yuqi_sequencing/SLX-18996_HeLa/bw .
rsync -arvuP martin03@10.20.192.94:/mnt/nfs/nas/group_folders/simeon01/Data/20200205_yuqi_sequencing/SLX-18996_HeLa/macs2_individual_rep .
```

Now including the peak calling scores:

```bash
##########
# HEK293 #
##########

cd /scratchb/sblab/martin03/repository/20200217_yuqi/data/20200217/bg4/SLX-19000_Hek/macs2_individual_rep

# biorep1
wc -l Hek_biorep1*.narrowPeak
  #  6890 Hek_biorep1_ChIP1_SLX-19000.i701_i502.r_1.hg38.q005.all_peaks.narrowPeak
  #  8584 Hek_biorep1_ChIP2_SLX-19000.i702_i502.r_1.hg38.q005.all_peaks.narrowPeak
  # 10165 Hek_biorep1_ChIP3_SLX-19000.i703_i502.r_1.hg38.q005.all_peaks.narrowPeak

tableCat.py -i Hek_biorep1*.narrowPeak | \
awk -v OFS="\t" '{print $1, $2, $3, $5, $NF}' | \
sort -k1,1 -k2,2n | \
bedtools merge -c 4,5 -o mean,count_distinct -i - | \
awk -v OFS="\t" '$5 > 1 {print $1, $2, $3, $4}' > Hek_biorep1_score.bed

wc -l Hek_biorep1_score.bed # 6957


# biorep2
wc -l Hek_biorep2*.narrowPeak
  # 15969 Hek_biorep2_ChIP1_SLX-19000.i701_i503.r_1.hg38.q005.all_peaks.narrowPeak
  # 12110 Hek_biorep2_ChIP2_SLX-19000.i702_i503.r_1.hg38.q005.all_peaks.narrowPeak
  #  9884 Hek_biorep2_ChIP3_SLX-19000.i703_i503.r_1.hg38.q005.all_peaks.narrowPeak

tableCat.py -i Hek_biorep2*.narrowPeak | \
awk -v OFS="\t" '{print $1, $2, $3, $5, $NF}' | \
sort -k1,1 -k2,2n | \
bedtools merge -c 4,5 -o mean,count_distinct -i - | \
awk -v OFS="\t" '$5 > 1 {print $1, $2, $3, $4}' > Hek_biorep2_score.bed

wc -l Hek_biorep2_score.bed # 10348


# biorep3
wc -l Hek_biorep3*.narrowPeak
  # 14524 Hek_biorep3_ChIP1_SLX-19000.i701_i504.r_1.hg38.q005.all_peaks.narrowPeak
  # 15506 Hek_biorep3_ChIP2_SLX-19000.i702_i504.r_1.hg38.q005.all_peaks.narrowPeak
  # 12541 Hek_biorep3_ChIP3_SLX-19000.i703_i504.r_1.hg38.q005.all_peaks.narrowPeak

tableCat.py -i Hek_biorep3*.narrowPeak | \
awk -v OFS="\t" '{print $1, $2, $3, $5, $NF}' | \
sort -k1,1 -k2,2n | \
bedtools merge -c 4,5 -o mean,count_distinct -i - | \
awk -v OFS="\t" '$5 > 1 {print $1, $2, $3, $4}' > Hek_biorep3_score.bed

wc -l Hek_biorep3_score.bed # 11679


# all biorep
wc -l Hek_biorep1_score.bed Hek_biorep2_score.bed Hek_biorep3_score.bed
 #  6957 Hek_biorep1_score.bed
 # 10348 Hek_biorep2_score.bed
 # 11679 Hek_biorep3_score.bed

tableCat.py -i Hek_biorep1_score.bed Hek_biorep2_score.bed Hek_biorep3_score.bed | \
sort -k1,1 -k2,2n | \
bedtools merge -c 4,5 -o mean,count_distinct -i - | \
awk -v OFS="\t" '$5 > 1 {print $1, $2, $3, $4}' > Hek_broad_score.bed

wc -l Hek_broad_score.bed # 9180
sort -k 4,4 Hek_broad_score.bed | head
# chr21	8442316	8442416	345.8888889
# chr21	8435709	8435887	345.7777778
# chr7	152463271	152464461	318.4444445
# chr21	8435112	8435221	317.2222222
# chr21	36156210	36156962	312.5555555
# chr21	8434847	8435033	306.8888889
# chr9	136790647	136791494	305.4444445
# chr1	244047929	244048791	305.4444444
# chr17	82022656	82023460	295.4444444
# chr10	132537140	132537953	285.7777778
```


### PQS

Given by Yuqi:

```bash
cd /scratchb/sblab/martin03/repository/20200217_yuqi/data/20200217/pqs
ls -lh
# -rwxrwxrwx 1 martin03 sblab 64M Sep 11 16:52 hg19.g3l12.bed
# -rwxrwxrwx 1 martin03 sblab 30M Sep 11 16:50 hg19.g3l7.bed

wc -l hg19.g3l12.bed hg19.g3l7.bed
  # 705580 hg19.g3l12.bed
  # 361424 hg19.g3l7.bed

sed 's/$/&\tPQS_G3L7/' hg19.g3l7.bed > hg19.g3l7.mod.bed
sed 's/$/&\tPQS_G3L12/' hg19.g3l12.bed > hg19.g3l12.mod.bed
```

Also map G3L7 and G3L12 in hg38:

```bash
srun --mem 64G --pty /usr/bin/bash

cd /scratchb/sblab/martin03/repository/20200217_yuqi/data/20200217/pqs

ref=/scratcha/sblab/martin03/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg38/Sequence/WholeGenomeFasta/genome.fa

nohup fastaRegexFinder.py -f $ref -q | \
bedtools sort -i | \
sed 's/$/&\tG3L7/' > hg38.g3l7.bed &

nohup fastaRegexFinder.py -f $ref -r '([gG]{3,}\w{1,12}){3,}[gG]{3,}' -q | \
bedtools sort -i | \
sed 's/$/&\tG3L12/' > hg38.g3l12.bed &

wc -l hg38*
   # 711954 hg38.g3l12.bed
   # 364468 hg38.g3l7.bed

exit
```



### OQS

#### HEK293T

Marsico2019 [GSE110582](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE110582)

- GSM3003539_Homo_all_w15_th-1_plus.hits.max.K.w50.25.bed.gz
- GSM3003539_Homo_all_w15_th-1_minus.hits.max.K.w50.25.bed.gz

```bash
cd /scratchb/sblab/martin03/repository/20200217_yuqi/data/20200217/oqs

wget ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3003nnn/GSM3003539/suppl/GSM3003539_Homo_all_w15_th-1_plus.hits.max.K.w50.25.bed.gz
wget ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3003nnn/GSM3003539/suppl/GSM3003539_Homo_all_w15_th-1_minus.hits.max.K.w50.25.bed.gz

pigz -d GSM3003539_Homo_all_w15_th-1_plus.hits.max.K.w50.25.bed.gz
pigz -d GSM3003539_Homo_all_w15_th-1_minus.hits.max.K.w50.25.bed.gz

wc -l GSM3003539_Homo_all_w15_th-1_*
  # 216321 GSM3003539_Homo_all_w15_th-1_minus.hits.max.K.w50.25.bed
  # 217951 GSM3003539_Homo_all_w15_th-1_plus.hits.max.K.w50.25.bed

bedtools merge -i <(cat GSM3003539_Homo_all_w15_th-1_* | bedtools sort -i) | \
sed 's/$/&\tOQS_HEK293T/' > hg19_oqs_hek293t.bed

wc -l hg19_oqs_hek293t.bed # 428624


# Convert to hg38
liftOver hg19_oqs_hek293t.bed /home/martin03/sw/liftOver/over_chain/hg19ToHg38.over.chain.gz hg38_oqs_hek293t.bed hg38_oqs_hek293t.unmapped.bed
wc -l hg38_oqs_hek293t.bed hg38_oqs_hek293t.unmapped.bed
  # 428186 hg38_oqs_hek293t.bed
  #    876 hg38_oqs_hek293t.unmapped.bed
```


#### NA18507

Chambers2015 [GSE63874](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE63874)

- GSE63874_Na_K_plus_hits_intersect.bed.gz
- GSE63874_Na_K_minus_hits_intersect.bed.gz

```bash
cd /scratchb/sblab/martin03/repository/20200217_yuqi/data/20200217/oqs

wget ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE63nnn/GSE63874/suppl/GSE63874_Na_K_plus_hits_intersect.bed.gz
wget ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE63nnn/GSE63874/suppl/GSE63874_Na_K_minus_hits_intersect.bed.gz

pigz -d GSE63874_Na_K_plus_hits_intersect.bed.gz
pigz -d GSE63874_Na_K_minus_hits_intersect.bed.gz

wc -l GSE63874_Na_K*
  # 260449 GSE63874_Na_K_minus_hits_intersect.bed
  # 265459 GSE63874_Na_K_plus_hits_intersect.bed

bedtools merge -i <(cat GSE63874_Na_K_* | bedtools sort -i) | \
sed 's/$/&\tOQS_NA18507/' > hg19_oqs_na18507.bed

wc -l hg19_oqs_na18507.bed # 496096


# Convert to hg38
liftOver hg19_oqs_na18507.bed /home/martin03/sw/liftOver/over_chain/hg19ToHg38.over.chain.gz hg38_oqs_na18507.bed hg38_oqs_na18507.unmapped.bed
wc -l hg38_oqs_na18507.bed hg38_oqs_na18507.unmapped.bed
  # 495519 hg38_oqs_na18507.bed
  #   1154 hg38_oqs_na18507.unmapped.bed
```



### Open chromatin

- ATAC-seq bigwig files available in [GSE108513](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE108513).
- DNase-seq peak files are available in [ENCSR000EJR](https://www.encodeproject.org/experiments/ENCSR000EJR/)

```bash
cd /scratchb/sblab/martin03/repository/20200217_yuqi/data/20200217/atac
wget ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2902nnn/GSM2902624/suppl/GSM2902624_HEK293_ATAC_high_depth_bio1_tech1.bw
wget ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2902nnn/GSM2902625/suppl/GSM2902625_HEK293_ATAC_high_depth_bio1_tech2.bw

cd /scratchb/sblab/martin03/repository/20200217_yuqi/data/20200217/dnase
wget ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2902nnn/GSM2902639/suppl/GSM2902639_HEK293_DNase.bw
wget https://www.encodeproject.org/files/ENCFF127KSH/@@download/ENCFF127KSH.bed.gz
zcat ENCFF127KSH.bed.gz | wc -l # 226744
pigz -d ENCFF127KSH.bed.gz
```



### RNA-seq

There are two possible datasets:

- [GSE106476](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE106476): see Negative datasets
- [GSE85161](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE85161): see polyA_plus_HEK_ctrl and polyA_minus_HEK_ctrl datasets

We will use `GSE106476` generated by Clemens, Barbara and Dhaval

GSE | GSM name | SRX | SRR
----|-----|-----|-----
GSE106476 | GSM2838596	Negative_1 | SRX3362136 | SRR6255657
GSE106476 | GSM2838597	Negative_2 | SRX3362137 | SRR6255658
GSE106476 | GSM2838598  Negative_3 | SRX3362138 | SRR6255659


#### Download fastq files

```bash
cd /scratchb/sblab/martin03/repository/20200217_yuqi/data/20200217

mkdir -p rnaseq/fastq && cd rnaseq/fastq

for run_id in SRR6255657 SRR6255658 SRR6255659
do
  sbatch -J $run_id -o $run_id.log --mem 16G --wrap "fastq-dump --outdir . --gzip --split-files -F $run_id"
done

tail *.log
```


#### Rename fastq files

```bash
cd /scratchb/sblab/martin03/repository/20200217_yuqi/data/20200217/rnaseq/fastq

# GSE106476 | GSM2838596	Negative_1 | SRX3362136 | SRR6255657
for filename in SRR6255657_{1,2}.fastq.gz; do mv "$filename" "Negative_1_$filename"; done;

# GSE106476 | GSM2838597	Negative_2 | SRX3362137 | SRR6255658
for filename in SRR6255658_{1,2}.fastq.gz; do mv "$filename" "Negative_2_$filename"; done;

# GSE106476 | GSM2838598  Negative_3 | SRX3362138 | SRR6255659
for filename in SRR6255659_{1,2}.fastq.gz; do mv "$filename" "Negative_3_$filename"; done;
```


#### Quality check

```bash
cd /scratchb/sblab/martin03/repository/20200217_yuqi/data/20200217/rnaseq/fastq

mkdir ../fastqc

for fq in *.fastq.gz
do
  bname=${fq%.fastq.gz}
  sbatch -J $bname -o ../fastqc/$bname.log --mem 4G --wrap "fastqc --noextract --nogroup -q -o ../fastqc $fq"
done
```

Get .html reports to `C02Q70MUFVH8`:

```bash
cd /Users/martin03/Desktop
mkdir fastqc
cd fastqc
rsync -arvuP martin03@10.20.236.34:/scratchb/sblab/martin03/repository/20200217_yuqi/data/20200217/rnaseq/fastqc/*.html .
```

Quality looks good



#### Trim illumina adapters and quality trimming

```bash
cd /scratchb/sblab/martin03/repository/20200217_yuqi/data/20200217/rnaseq/fastq

mkdir ../fastq_trimmed

for fq1 in Negative_*_1.fastq.gz
do
  fq2=${fq1/_1./_2.}
  bname=${fq1%_1.fastq.gz}
  #echo $fq1, $fq2, $bname
  sbatch -J $bname -o ../fastq_trimmed/$bname.log --mem 4G --wrap "cutadapt -a AGATCGGAAGAGC -A AGATCGGAAGAGC -m 15 -q 20 -o ../fastq_trimmed/$fq1 -p ../fastq_trimmed/$fq2 $fq1 $fq2 > ../fastq_trimmed/$bname.txt"
done

cd ../fastq_trimmed

grep "Read 1 with adapter" *.txt
# Negative_1_SRR6255657.txt:  Read 1 with adapter:                 505,736 (1.0%)
# Negative_2_SRR6255658.txt:  Read 1 with adapter:                 502,497 (1.0%)
# Negative_3_SRR6255659.txt:  Read 1 with adapter:                 492,547 (1.0%)

grep "Read 2 with adapter" *.txt
# Negative_1_SRR6255657.txt:  Read 2 with adapter:                 744,627 (1.4%)
# Negative_2_SRR6255658.txt:  Read 2 with adapter:                 738,256 (1.4%)
# Negative_3_SRR6255659.txt:  Read 2 with adapter:                 691,583 (1.4%)

grep "Pairs written (passing filters):" *.txt
# Negative_1_SRR6255657.txt:Pairs written (passing filters):    52,660,353 (99.8%)
# Negative_2_SRR6255658.txt:Pairs written (passing filters):    50,871,581 (99.7%)
# Negative_3_SRR6255659.txt:Pairs written (passing filters):    48,039,958 (99.7%)
```


#### Alignment

##### Prepare rsem reference sequences

```bash
srun --mem 64G --pty /usr/bin/bash

cd /scratcha/sblab/martin03/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg38/Sequence

mkdir rsem_bowtie2

# bowtie2
nohup rsem-prepare-reference --gtf /scratcha/sblab/martin03/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg38/Annotation/Genes/genes.gtf \
--bowtie2 \
-p 20 \
/scratcha/sblab/martin03/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg38/Sequence/WholeGenomeFasta/genome.fa \
/scratcha/sblab/martin03/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg38/Sequence/rsem_bowtie2/genome &

tail nohup.out
rm nohup.out

exit
```

However the `rsem-calculate-expression` below did not work with `hg38` so I went back to using `GRCh38.p12.genome` like before.


##### Align

```bash
cd /scratchb/sblab/martin03/repository/20200217_yuqi/data/20200217/rnaseq/fastq_trimmed

mkdir ../rsem

bwtidx_human=/scratcha/sblab/martin03/reference_data/genomes/gencode/Gencode_human/release_28/rsem_bowtie2_basic/GRCh38.p12.genome

for fq1 in *_1.fastq.gz
do
  fq2=${fq1/_1./_2.}
  bname=${fq1%_1.fastq.gz}
  echo $fq1, $fq2, $bname
  sbatch -J $bname -o ../rsem/$bname.log --mem 32G --wrap "rsem-calculate-expression \
  --paired-end \
  -p 20 \
  --bowtie2 \
  --append-names \
  $fq1 \
  $fq2 \
  $bwtidx_human \
  ../rsem/$bname"
done

tail ../rsem/*.log

cd ../rsem

grep "overall alignment rate" *.log
# Negative_1_SRR6255657.log:68.25% overall alignment rate
# Negative_2_SRR6255658.log:64.86% overall alignment rate
# Negative_3_SRR6255659.log:60.96% overall alignment rate

wc -l *.genes.results
   # 58382 Negative_1_SRR6255657.genes.results
   # 58382 Negative_2_SRR6255658.genes.results
   # 58382 Negative_3_SRR6255659.genes.results
```



### Promoters and link to RNA-seq

Using GenomicFeatures:

```bash
library(data.table)
library(GenomicFeatures)

# change width
options(width = 250)

# prepare coordinates table
txdb <- makeTxDbFromGFF("/scratcha/sblab/martin03/reference_data/genomes/gencode/Gencode_human/release_28/gtf/gencode.v28.annotation.gtf", format="gtf")

# gene promoters
gene_promoters <- data.table(data.frame(promoters(genes(txdb, columns="gene_id"), upstream=1000, downstream=0)))[, c("seqnames", "start", "end", "gene_id", "strand")][order(seqnames, start)]
gene_promoters[, start := ifelse(start < 0, 0, start)]
gene_promoters[, dummy := "."]
gene_promoters <- gene_promoters[, .(seqnames, start, end, gene_id, dummy, strand)]


# RNA-seq
## Negative_1
neg_1 <- fread("/scratchb/sblab/martin03/repository/20200217_yuqi/data/20200217/rnaseq/rsem/Negative_1_SRR6255657.genes.results")
neg_1[, ensembl_id := sapply(neg_1$gene_id, function(x) unlist(strsplit(x, "_"))[1])]
neg_1[, gene_name := sapply(neg_1$gene_id, function(x) unlist(strsplit(x, "_"))[2])]
neg_1 <- neg_1[, c("ensembl_id", "gene_name", "TPM")]

## Negative_2
neg_2 <- fread("/scratchb/sblab/martin03/repository/20200217_yuqi/data/20200217/rnaseq/rsem/Negative_2_SRR6255658.genes.results")
neg_2[, ensembl_id := sapply(neg_2$gene_id, function(x) unlist(strsplit(x, "_"))[1])]
neg_2[, gene_name := sapply(neg_2$gene_id, function(x) unlist(strsplit(x, "_"))[2])]
neg_2 <- neg_2[, c("ensembl_id", "gene_name", "TPM")]

## Negative_3
neg_3 <- fread("/scratchb/sblab/martin03/repository/20200217_yuqi/data/20200217/rnaseq/rsem/Negative_3_SRR6255659.genes.results")
neg_3[, ensembl_id := sapply(neg_3$gene_id, function(x) unlist(strsplit(x, "_"))[1])]
neg_3[, gene_name := sapply(neg_3$gene_id, function(x) unlist(strsplit(x, "_"))[2])]
neg_3 <- neg_3[, c("ensembl_id", "gene_name", "TPM")]


# Merge
setkey(gene_promoters, "gene_id")
setkey(neg_1, "ensembl_id")
setkey(neg_2, "ensembl_id")
setkey(neg_3, "ensembl_id")

gene_promoters <- gene_promoters[neg_1,][neg_2,][neg_3,][order(seqnames, start)]
gene_promoters[, TPM_mean := (TPM + i.TPM + i.TPM.1)/3]
gene_promoters[, TPM_mean := round(TPM_mean, 2)]
gene_promoters <- gene_promoters[, c("seqnames", "start", "end", "gene_id", "gene_name", "strand", "TPM_mean")]
summary(gene_promoters$TPM_mean)
     # Min.   1st Qu.    Median      Mean   3rd Qu.      Max.
     # 0.00      0.00      0.03     17.06      0.65 124200.00

write.table(gene_promoters, file = "/scratchb/sblab/martin03/repository/20200217_yuqi/data/20200217/rnaseq/rsem/genepromoters_rnaseq.bed", row.names = FALSE, col.names = FALSE, sep = '\t', quote = FALSE)
```



### Overlaps

```bash
cd /scratchb/sblab/martin03/repository/20200217_yuqi/data/20200217/

wc -l rnaseq/rsem/genepromoters_rnaseq.bed bg4/SLX-19000_Hek/macs2_individual_rep/Hek_broad_score.bed pqs/hg38.g3l7.bed oqs/hg38_oqs_hek293t.bed
  #  58651 rnaseq/rsem/genepromoters_rnaseq.bed
  #   9180 bg4/SLX-19000_Hek/macs2_individual_rep/Hek_broad_score.bed
  # 364468 pqs/hg38.g3l7.bed
  # 428186 oqs/hg38_oqs_hek293t.bed


# Promoters + BG4
bedtools intersect \
-a rnaseq/rsem/genepromoters_rnaseq.bed \
-b bg4/SLX-19000_Hek/macs2_individual_rep/Hek_broad_score.bed \
-wa -wb > rnaseq/rsem/genepromoters_rnaseq_Hek_broad_score.bed

wc -l rnaseq/rsem/genepromoters_rnaseq_Hek_broad_score.bed # 8233
head rnaseq/rsem/genepromoters_rnaseq_Hek_broad_score.bed | column -t
# chr1  633376   634375   ENSG00000198744.5   RP5-857K21.11  +  0.07   chr1  633985   634083   203.4444445
# chr1  777770   778769   ENSG00000237491.8   RP11-206L10.9  +  5.67   chr1  778533   778951   90.91666667
# chr1  778627   779626   ENSG00000228327.3   RP11-206L10.2  -  0.78   chr1  778533   778951   90.91666667
# chr1  827523   828522   ENSG00000225880.5   NCRNA00115     -  0.71   chr1  827430   827601   76.5
# chr1  903834   904833   ENSG00000272438.1   RP11-54O7.16   +  0.14   chr1  904601   904928   49.66666667
# chr1  959310   960309   ENSG00000188976.10  NOC2L          -  55.85  chr1  959133   959478   83.77777777
# chr1  959587   960586   ENSG00000187961.13  KLHL17         +  4.22   chr1  960417   960717   57.5
# chr1  1000138  1001137  ENSG00000187608.8   ISG15          +  3.49   chr1  999983   1001005  93.46111111
# chr1  1000173  1001172  ENSG00000188290.10  HES4           -  59.92  chr1  999983   1001005  93.46111111
# chr1  1019123  1020122  ENSG00000188157.14  AGRN           +  3.87   chr1  1019250  1020176  76.11111111


# Promoters + BG4 + G3L7
bedtools intersect \
-a rnaseq/rsem/genepromoters_rnaseq_Hek_broad_score.bed \
-b pqs/hg38.g3l7.bed \
-wa -wb > rnaseq/rsem/genepromoters_rnaseq_Hek_broad_score_hg38.g3l7.bed

wc -l rnaseq/rsem/genepromoters_rnaseq_Hek_broad_score_hg38.g3l7.bed # 8898
head rnaseq/rsem/genepromoters_rnaseq_Hek_broad_score_hg38.g3l7.bed | column -t
# chr1  633376   634375   ENSG00000198744.5   RP5-857K21.11  +  0.07   chr1  633985  634083   203.4444445  chr1  633420   633464   chr1_633420_633464_rev    44  -  CCCGTATTTACCCTATAGCACCCCCTCTACCCCCTCTAGAGCCC      G3L7
# chr1  827523   828522   ENSG00000225880.5   NCRNA00115     -  0.71   chr1  827430  827601   76.5         chr1  827586   827629   chr1_827586_827629_for    43  +  ggggccgggccgggccggggcggggtctcgggcaggggcgggg       G3L7
# chr1  827523   828522   ENSG00000225880.5   NCRNA00115     -  0.71   chr1  827430  827601   76.5         chr1  827986   828027   chr1_827986_828027_for    41  +  GGGGTCGGGCCCAGGGCTGGAGGGGCCGGGACCGGGCGGGG         G3L7
# chr1  903834   904833   ENSG00000272438.1   RP11-54O7.16   +  0.14   chr1  904601  904928   49.66666667  chr1  904022   904055   chr1_904022_904055_rev    33  -  CCCCTGCAGACCCTGTGCCCAGCCCAGCATCCC                 G3L7
# chr1  903834   904833   ENSG00000272438.1   RP11-54O7.16   +  0.14   chr1  904601  904928   49.66666667  chr1  904114   904140   chr1_904114_904140_for    26  +  GGGAGGCCTGGGGCGGAGGGCCGGGG                        G3L7
# chr1  903834   904833   ENSG00000272438.1   RP11-54O7.16   +  0.14   chr1  904601  904928   49.66666667  chr1  904296   904325   chr1_904296_904325_for    29  +  GGGTGGGGGGGTGGGGGGCGGCATCGGGG                     G3L7
# chr1  959310   960309   ENSG00000188976.10  NOC2L          -  55.85  chr1  959133  959478   83.77777777  chr1  959307   959334   chr1_959307_959334_rev    27  -  cccaggccccgcccctttccccgcccc                       G3L7
# chr1  959310   960309   ENSG00000188976.10  NOC2L          -  55.85  chr1  959133  959478   83.77777777  chr1  959521   959544   chr1_959521_959544_rev    23  -  CCCGCCGCGCCCGCAGCCCACCC                           G3L7
# chr1  1000138  1001137  ENSG00000187608.8   ISG15          +  3.49   chr1  999983  1001005  93.46111111  chr1  1000362  1000410  chr1_1000362_1000410_for  48  +  GGGAGGGGCCGCTGGGGGCGAACGGGGCCCGGGACCCCCGGGGCTGGG  G3L7
# chr1  1000173  1001172  ENSG00000188290.10  HES4           -  59.92  chr1  999983  1001005  93.46111111  chr1  1000362  1000410  chr1_1000362_1000410_for  48  +  GGGAGGGGCCGCTGGGGGCGAACGGGGCCCGGGACCCCCGGGGCTGGG  G3L7



# Promoters + BG4 + G3L7 + OQS
bedtools intersect \
-a rnaseq/rsem/genepromoters_rnaseq_Hek_broad_score_hg38.g3l7.bed \
-b oqs/hg38_oqs_hek293t.bed \
-wa -wb > rnaseq/rsem/genepromoters_rnaseq_Hek_broad_score_hg38.g3l7_hg38_oqs_hek293t.bed

wc -l rnaseq/rsem/genepromoters_rnaseq_Hek_broad_score_hg38.g3l7_hg38_oqs_hek293t.bed # 12059
head rnaseq/rsem/genepromoters_rnaseq_Hek_broad_score_hg38.g3l7_hg38_oqs_hek293t.bed | column -t
# chr1  633376  634375  ENSG00000198744.5   RP5-857K21.11  +  0.07   chr1  633985  634083  203.4444445  chr1  633420  633464  chr1_633420_633464_rev  44  -  CCCGTATTTACCCTATAGCACCCCCTCTACCCCCTCTAGAGCCC  G3L7  chr1  633360  633480  OQS_HEK293T
# chr1  827523  828522  ENSG00000225880.5   NCRNA00115     -  0.71   chr1  827430  827601  76.5         chr1  827586  827629  chr1_827586_827629_for  43  +  ggggccgggccgggccggggcggggtctcgggcaggggcgggg   G3L7  chr1  827556  827706  OQS_HEK293T
# chr1  827523  828522  ENSG00000225880.5   NCRNA00115     -  0.71   chr1  827430  827601  76.5         chr1  827986  828027  chr1_827986_828027_for  41  +  GGGGTCGGGCCCAGGGCTGGAGGGGCCGGGACCGGGCGGGG     G3L7  chr1  827556  827706  OQS_HEK293T
# chr1  903834  904833  ENSG00000272438.1   RP11-54O7.16   +  0.14   chr1  904601  904928  49.66666667  chr1  904022  904055  chr1_904022_904055_rev  33  -  CCCCTGCAGACCCTGTGCCCAGCCCAGCATCCC             G3L7  chr1  904267  904486  OQS_HEK293T
# chr1  903834  904833  ENSG00000272438.1   RP11-54O7.16   +  0.14   chr1  904601  904928  49.66666667  chr1  904114  904140  chr1_904114_904140_for  26  +  GGGAGGCCTGGGGCGGAGGGCCGGGG                    G3L7  chr1  904267  904486  OQS_HEK293T
# chr1  903834  904833  ENSG00000272438.1   RP11-54O7.16   +  0.14   chr1  904601  904928  49.66666667  chr1  904296  904325  chr1_904296_904325_for  29  +  GGGTGGGGGGGTGGGGGGCGGCATCGGGG                 G3L7  chr1  904267  904486  OQS_HEK293T
# chr1  959310  960309  ENSG00000188976.10  NOC2L          -  55.85  chr1  959133  959478  83.77777777  chr1  959307  959334  chr1_959307_959334_rev  27  -  cccaggccccgcccctttccccgcccc                   G3L7  chr1  959220  959340  OQS_HEK293T
# chr1  959310  960309  ENSG00000188976.10  NOC2L          -  55.85  chr1  959133  959478  83.77777777  chr1  959307  959334  chr1_959307_959334_rev  27  -  cccaggccccgcccctttccccgcccc                   G3L7  chr1  960060  960675  OQS_HEK293T
# chr1  959310  960309  ENSG00000188976.10  NOC2L          -  55.85  chr1  959133  959478  83.77777777  chr1  959521  959544  chr1_959521_959544_rev  23  -  CCCGCCGCGCCCGCAGCCCACCC                       G3L7  chr1  959220  959340  OQS_HEK293T
# chr1  959310  960309  ENSG00000188976.10  NOC2L          -  55.85  chr1  959133  959478  83.77777777  chr1  959521  959544  chr1_959521_959544_rev  23  -  CCCGCCGCGCCCGCAGCCCACCC                       G3L7  chr1  960060  960675  OQS_HEK293T

sort -k11,11nr rnaseq/rsem/genepromoters_rnaseq_Hek_broad_score_hg38.g3l7_hg38_oqs_hek293t.bed | head
```



### Fig1d Robert's Nature Genetics paper

```bash
cd /scratchb/sblab/martin03/repository/20200217_yuqi/data/20200217

wc -l rnaseq/rsem/genepromoters_rnaseq.bed bg4/SLX-19000_Hek/macs2_individual_rep/Hek_broad.bed dnase/ENCFF127KSH.bed oqs/hg38_oqs_hek293t.bed
  #  58651 rnaseq/rsem/genepromoters_rnaseq.bed
  #   9337 bg4/SLX-19000_Hek/macs2_individual_rep/Hek_broad.bed
  # 226744 dnase/ENCFF127KSH.bed
  # 428186 oqs/hg38_oqs_hek293t.bed


# Promoters + DNase
bedtools intersect \
-a rnaseq/rsem/genepromoters_rnaseq.bed \
-b dnase/ENCFF127KSH.bed \
-wa -u > rnaseq/rsem/genepromoters_rnaseq_ENCFF127KSH.bed

wc -l rnaseq/rsem/genepromoters_rnaseq_ENCFF127KSH.bed # 18420


# Promoters + DNase + OQs
bedtools intersect \
-a rnaseq/rsem/genepromoters_rnaseq_ENCFF127KSH.bed \
-b oqs/hg38_oqs_hek293t.bed \
-wa -u > rnaseq/rsem/genepromoters_rnaseq_ENCFF127KSH_hg38_oqs_hek293t.bed

wc -l rnaseq/rsem/genepromoters_rnaseq_ENCFF127KSH_hg38_oqs_hek293t.bed # 9019


# Promoters + DNase + OQs +/- BG4
cat \
<(bedtools intersect \
-a rnaseq/rsem/genepromoters_rnaseq_ENCFF127KSH_hg38_oqs_hek293t.bed \
-b bg4/SLX-19000_Hek/macs2_individual_rep/Hek_broad.bed \
-wa -u | sed 's/$/&\tbg4/') \
<(bedtools intersect \
-a rnaseq/rsem/genepromoters_rnaseq_ENCFF127KSH_hg38_oqs_hek293t.bed \
-b bg4/SLX-19000_Hek/macs2_individual_rep/Hek_broad.bed \
-v | sed 's/$/&\tnobg4/') | \
bedtools sort -i > rnaseq/rsem/genepromoters_rnaseq_ENCFF127KSH_hg38_oqs_hek293t_Hek_broad.bed

wc -l rnaseq/rsem/genepromoters_rnaseq_ENCFF127KSH_hg38_oqs_hek293t_Hek_broad.bed # 9019
cut -f8 rnaseq/rsem/genepromoters_rnaseq_ENCFF127KSH_hg38_oqs_hek293t_Hek_broad.bed | sort | uniq -c
   # 4034 bg4
   # 4985 nobg4
```

Boxplot:

```r
# cd /scratchb/sblab/martin03/repository/20200217_yuqi/data/20200217

library(data.table)
library(ggplot2)
library(ggpubr)

# Enlarge the view width when printing tables
options(width = 350)

# Load metadata
data <- fread("/scratchb/sblab/martin03/repository/20200217_yuqi/data/20200217/rnaseq/rsem/genepromoters_rnaseq_ENCFF127KSH_hg38_oqs_hek293t_Hek_broad.bed")
setnames(data, c("chr", "start", "end", "ensembl_id", "gene_id", "strand", "tpm", "class"))

mean(data[class == "bg4"]$tpm) # 39.36073
mean(data[class == "nobg4"]$tpm) # 16.30799


# Boxplot
gg <- ggplot(data = data, aes(x = factor(class, levels = c("nobg4", "bg4")), y = log10(tpm))) +
geom_boxplot(outlier.shape=NA) +
ylab(expression("log"[10]*"TPM")) +
xlab(expression("")) +
theme_classic() +
ggtitle("DNase (+) and OQs (+)") +
scale_x_discrete(labels=c("bg4" = sprintf("G4\n(%s)", as.character(table(data$class)[1])), "nobg4" = sprintf("no G4\n(%s)", as.character(table(data$class)[2])))) +
stat_compare_means(comparisons = list(c("bg4", "nobg4")), method = "wilcox.test", label.y = 3.75) +
theme(axis.title = element_text(size=16), axis.text = element_text(size=16, color = "black"), strip.text = element_text(size=16, color = "black"), plot.title = element_text(size=16, hjust = 0.5)) +
coord_cartesian(ylim = c(-2, 4))

ggsave("/Users/martin03/github/sblab-bioinformatics/projects/20200217_yuqi/figures/20200316_hek_fig1d.pdf", width = 5, height = 6)
```



### ENCODE

Following [Jochen's script](https://github.com/sblab-bioinformatics/projects/blob/master/20171123_Jochen_ENCODE/Shuffling-Analysis/Additional%20Cell%20lines/HEK293_Analysis.md).

We are interested in the [216 TF ChIP-seq experiments](https://www.encodeproject.org/search/?type=Experiment&status=released&biosample_ontology.term_name=HEK293&assay_title=TF+ChIP-seq) as available on the 18 February 2020.

The metadata was downloaded from [here](https://www.encodeproject.org/metadata/?type=Experiment&status=released&biosample_ontology.term_name=HEK293&assay_title=TF+ChIP-seq)

### Filter metadata

```r
library(data.table)

options(width = 300)

# Load metadata
data <- data.table(read.table(file = "/scratchb/sblab/martin03/repository/20200217_yuqi/data/20200217/encode/metadata_hek293.tsv", sep = '\t', header = TRUE))

# Filter metadata
data_filtered <- data[(File.format == 'bed narrowPeak' | File.format == 'bed broadPeak') & Output.type == 'optimal IDR thresholded peaks' & Assembly == 'GRCh38' & File.Status == 'released']

# Write
write.table(data_filtered, "/scratchb/sblab/martin03/repository/20200217_yuqi/data/20200217/encode/metadata_hek293_filtered.tsv", sep="\t", quote = FALSE, row.names = FALSE, col.names = FALSE)
```





## Clean-up

```bash
cd /scratchb/sblab/martin03/repository/20200217_yuqi/data
rm -r 20200217
```
